### DISCUSSION TITLE
Understanding of Twig filters

### DISCUSSION THREAD

**Discussion by svonlanthen** [craftcms/cms#12104] [Q&A] [answered]
Hi everyone

Just that i can understand:
Why is symphony/twig included in craft, but not possible to use all Twig Filters?

For example: 
I want to use the Twig-Filter https://twig.symfony.com/doc/3.x/filters/format_datetime.html 
But it's not available in https://craftcms.com/docs/4.x/dev/filters.html 

Craft CMS 4.2.7
Twig 3.4
PHP 8.1.11

Thank you very much!

**Comment by brianjhanson** [comment, marked as answer]
The `format_datetime` filter is a part of the `IntlExtension` which isn't installed in Craft by default. This would also apply to [all the filters added by the package](https://github.com/twigphp/intl-extra). 

There's a small note about that at the very bottom of the documentation in Symfony (admittedly very easy to miss). 

If you'd like to use the filters provided by the extension, you can composer install it into your project and [register the twig extension in a plugin or module](https://craftcms.com/docs/4.x/extend/extending-twig.html#register-a-twig-extension).

**Reply by svonlanthen** [reply]
@brianjhanson Thank you very much for your reply 🌻🌼🌷

**Reply by brandonkelly** [reply]
Worth noting that Craft provides a custom [`|date` filter](https://craftcms.com/docs/4.x/dev/filters.html#date), which has the same functionality provided by `|format_datetime` built-in. For example you can do

```twig
{{ entry.postDate|date('short') }}
```

to output a locale-aware short date format.

**Comment by svonlanthen** [comment]
Dear all

I got it: https://github.com/craftcms/cms/issues/4558#issuecomment-511224883

It may not answer my question, but i am happy with how it works.

It would be a great addition to the documentation (https://craftcms.com/docs/4.x/dev/filters.html#date) to mention the possibility to use the date filter like mentioned at https://github.com/craftcms/cms/issues/4558#issuecomment-511224883 .

Best regards
